syntax = "proto3";
package auth;

option java_package = "com.example.auth";
option go_package = "github.com/example/distributed-ticket-booking/proto/auth;auth";

// User registration and login messages
message RegisterRequest {
  string email = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message Session {
  string token = 1;
  string user_id = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
  Session session = 3;
}

// Session validation message
message ValidateSessionRequest {
  string token = 1;
}

message ValidateSessionResponse {
  bool valid = 1;
  string user_id = 2; // The authenticated user's ID
}

// Authentication Service definition
service AuthService {
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc ValidateSession(ValidateSessionRequest) returns (ValidateSessionResponse);
}